module_3/gift_certificates_system/model/src/main/java/com/epam/esm/entity/User.java

```diff
@@ -0,0 +1,52 @@
+package com.epam.esm.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * User entity.
+ */
+@RequiredArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Entity
+@ToString
+public class User implements Serializable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    private String login;
+
+    @Column(nullable = false)
+    private String firstName;
+
+    @Column(nullable = false)
+    private String lastName;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        User user = (User) o;
+        return getLogin().equals(user.getLogin());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getLogin());
+    }
```

Comment:
Why didn't you define this via lombok?

Reply:
Using `@EqualsAndHashCode` for JPA entities is not recommended. It can cause severe performance and memory consumption issues.